kore/src/Kore/Simplify/Not.hs

```diff
@@ -78,42 +78,16 @@ Right now this uses the following:
 -}
 simplify ::
     MonadSimplify simplifier =>
+    Ord sort =>
     SideCondition RewritingVariableName ->
+    --TODO (Andrei B): Sort vs sort
     Not sort (OrPattern RewritingVariableName) ->
     simplifier (OrPattern RewritingVariableName)
-simplify sideCondition Not{notChild} =
-    simplifyEvaluated sideCondition notChild
-
-{- |'simplifyEvaluated' simplifies a 'Not' pattern given its
-'OrPattern' child.
-
-See 'simplify' for details.
--}
-
-{- TODO (virgil): Preserve pattern sorts under simplification.
-
-One way to preserve the required sort annotations is to make 'simplifyEvaluated'
-take an argument of type
-
-> CofreeF (Not Sort) (Attribute.Pattern variable) (OrPattern variable)
-
-instead of an 'OrPattern' argument. The type of 'makeEvaluate' may
-be changed analogously. The 'Attribute.Pattern' annotation will eventually
-cache information besides the pattern sort, which will make it even more useful
-to carry around.
-
--}
-simplifyEvaluated ::
-    MonadSimplify simplifier =>
-    SideCondition RewritingVariableName ->
-    OrPattern RewritingVariableName ->
-    simplifier (OrPattern RewritingVariableName)
-simplifyEvaluated sideCondition simplified =
+simplify sideCondition not'@Not{notSort} =
```

Comment:
I think the caller should decide what to do with the sort here, so basically in the `Predicate` simplifier we should delete the sort there instead of rely on this function being parametric in the sort. I hope this makes sense, please let me know if it doesn't.

Reply:
I don't understand. If this function uses a concrete `Sort` type, wouldn't we need to provide the sort in the predicate simplifier?

Reply:
I mean, we use this because we're simplifying a term (like for the `Floor` case, its child is always a term) so we should be using that term's sort here.